source/Calamari.Tests/KubernetesFixtures/ResourceStatus/KubernetesYamlTests.cs

```diff
@@ -0,0 +1,31 @@
+using System.Linq;
+using Calamari.Kubernetes.ResourceStatus;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Calamari.Tests.KubernetesFixtures.ResourceStatus
+{
+    [TestFixture]
+    public class KubernetesYamlTests
+    {
+        [Test]
+        public void ShouldGenerateCorrectIdentifiers()
+        {
+            var input = TestFileLoader.Load("single-deployment.yaml");
+            var got = KubernetesYaml.GetDefinedResources(input);
+            got.Should().HaveCount(1);
+            var identifier = got.First();
+            identifier.Kind.Should().Be("Deployment");
+            identifier.Name.Should().Be("nginx");
+            identifier.Namespace.Should().Be("default");
+        }
+
+        [Test]
+        public void ShouldHandleMultipleResourcesDefinedInTheSameFile()
+        {
+            var input = TestFileLoader.Load("multiple-resources.yaml");
+            var got = KubernetesYaml.GetDefinedResources(input);
+            got.Should().HaveCount(3);
```

Comment:
I think we should also validate that we parsed the correct combo of type + name + namespace for each resource